flake.nix

```diff
@@ -21,34 +22,28 @@
           export RUSTC_WRAPPER=sccache
         '';
       };
-      packages.${system} = {
-        hyprland-workspaces = pkgs.rustPlatform.buildRustPackage {
-          name = "hyprland-workspaces";
-          src = ./hyprland-workspaces;
-          cargoLock = {
-            lockFile = ./hyprland-workspaces/Cargo.lock;
-          };
-          nativeBuildInputs = [];
-          buildInputs = [];
-        };
-        hyprland-current-window-title = pkgs.rustPlatform.buildRustPackage {
-          name = "hyprland-current-window-title";
-          src = ./hyprland-current-window-title;
-          cargoLock = {
-            lockFile = ./hyprland-current-window-title/Cargo.lock;
-          };
-          nativeBuildInputs = [];
-          buildInputs = [];
-        };
-        eww-launch = pkgs.rustPlatform.buildRustPackage {
-          name = "eww-launch";
-          src = ./eww-launch;
-          cargoLock = {
-            lockFile = ./eww-launch/Cargo.lock;
-          };
-          nativeBuildInputs = [];
-          buildInputs = [];
+      packages.${system} = let
+        targets = [
+          "eww-launch"
+          "hyprland-current-window-title"
+          "hyprland-workspaces"
+        ];
+        packages = builtins.foldl' (acc: t:
+          {
+            ${t} = pkgs.rustPlatform.buildRustPackage {
+              name = t;
+              src = ./.;
+              cargoBuildFlags = "--bin ${t}";
+              cargoLock.lockFile = ./Cargo.lock;
+            };
+          } // acc) { } targets;
+      in packages // rec {
+        all = pkgs.symlinkJoin {
+          name = "eww-scripts";
+          paths = builtins.attrValues packages;
```

Comment:
I am unsure whether there is a way to improve the usage of the `packages` binding as it seems a bit off to me. Doesn't really matter as long as it works though

Reply:
Might look into this at another point in time